modules/document_repository/jsx/categoryForm.js

```diff
@@ -0,0 +1,211 @@
+              ref="category_name"
+              required={true}
+              value={this.state.formData.category_name}
+            />
+            <SelectElement
+              name="parent_id"
+              label="Parent"
+              options={this.state.data.fieldOptions.fileCategories}
+              onUserInput={this.setFormData}
+              ref="parent_id"
+              hasError={false}
+              value={this.state.formData.parent_id}
+            />
+            <TextareaElement
+              name="comments"
+              label="Comments"
+              onUserInput={this.setFormData}
+              ref="comments"
+              value={this.state.formData.comments}
+            />
+            <ButtonElement label="Add Category"/>
+          </FormElement>
+        </div>
+      </div>
+    );
+  }
+
+/** *******************************************************************************
+ *                      ******     Helper methods     *******
+ *********************************************************************************/
+
+
+  /**
+   * Handle form submission
+   * @param {object} e - Form submission event
+   */
+  handleSubmit(e) {
+    e.preventDefault();
+
+    let formData = this.state.formData;
+    let formRefs = this.refs;
+
+    // Validate the form
+    if (!this.isValidForm(formRefs, formData)) {
+      return;
+    }
+      this.uploadFile();
+  }
+  /*
+   * Uploads the file to the server
+   */
+  uploadFile() {
+    // Set form data and upload the media file
+    let formData = this.state.formData;
+    let formObj = new FormData();
+    for (let key in formData) {
+      if (formData[key] !== '') {
+        formObj.append(key, formData[key]);
+      }
+    }
+   fetch(this.props.action, {
+      method: 'POST',
+      cache: 'no-cache',
+      credentials: 'same-origin',
+      body: formObj,
+    })
+    .then((resp) => resp.json())
+    .then(()=>{
+        this.props.refreshPage();
+        this.fetchData();
+        // refresh the upload page
+        this.props.newCategoryState();
+        this.setState({
+          formData: {}, // reset form data after successful file upload
+          uploadProgress: -1,
+        });
+        swal('Add Successful!', '', 'success');
+    });
+  }
+  /**
+   * Validate the form
+   *
+   * @param {object} formRefs - Object containing references to React form elements
+   * @param {object} formData - Object containing form data inputed by user
+   * @return {boolean} - true if all required fields are filled, false otherwise
+   */
+  isValidForm(formRefs, formData) {
+    let isValidForm = true;
+
+    let requiredFields = {
+      category: null,
+      site: null,
+      file: null,
+    };
+
+    Object.keys(requiredFields).map(function(field) {
+      if (formData[field]) {
+        requiredFields[field] = formData[field];
+      } else if (formRefs[field]) {
+        formRefs[field].props.hasError = true;
+        isValidForm = false;
+      }
```

Comment:
Refs should be used here. Error messages should be tracked via a error state object.